src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -27,8 +27,12 @@ protected CompilationModuleGroup(CompilerTypeSystemContext typeSystemContext, IC
             _rootProvider = rootProvider;
         }
 
-        public abstract bool IsTypeInCompilationGroup(TypeDesc type);
-        public abstract bool IsMethodInCompilationGroup(MethodDesc method);
+        public abstract bool IsInCompilationGroup(TypeDesc type);
+        public abstract bool IsInCompilationGroup(MethodDesc method);
+        public abstract bool ShouldShareAcrossModules(TypeDesc type);
```

Comment:
There is certainly a cost to using COMDAT sections. Instead of one section for all methods we create a new section for each method. I haven't measured it yet but I would prefer to do this in a follow up checkin if that's fine with you. This set of changes is large and annoying to keep resolving merge conflicts.